### PR TITLE
feat: publish rustdoc of main branch

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- update CI setup to publish rustdoc of the `main` branch to github pages
- update `readme.md` accordingly

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/forest/issues/1996

**Other information and links**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -157,3 +157,27 @@ jobs:
...
+  gh-pages:
+    env:
+      # https://doc.rust-lang.org/nightly/rustdoc/unstable-features.html#--index-page-provide-a-top-level-landing-page-for-docs
+      RUSTDOCFLAGS: "-Zunstable-options --enable-index-page"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Dependencies
+        if: startsWith(matrix.os, 'Ubuntu')
+        run: |
+          sudo apt-get update
```
Comment (hanabi1224) [inline .github/workflows/rust.yml:170]: How about moving the ubuntu deps setup to `./scripts/setup_ubuntu.sh`

**Comment by hanabi1224** [comment]
Hmm, looks like I'm DDOSing the CI build resources cuz current cancel workflow setup does not work for PRs from fork repos

**Review by lemmih — APPROVED** [review]

**Comment by hanabi1224** [comment]
@lemmih  Looks like I have to move the rustdoc link to another PR since it's not yet published and it's caught by the new readme.md linter  : )

File: README.md [hunk 2]
```diff
@@ -5,6 +5,8 @@
 [![GitHub Workflow Status](https://img.shields.io/github/workflow/status/ChainSafe/forest/Rust?style=for-the-badge)](https://github.com/ChainSafe/forest/actions)
 [![Codecov](https://img.shields.io/codecov/c/gh/ChainSafe/forest?style=for-the-badge&token=1OHO2CSD17)](https://codecov.io/gh/ChainSafe/forest)
 [![GitHub release (latest by date)](https://img.shields.io/github/v/release/ChainSafe/forest?style=for-the-badge)](https://github.com/ChainSafe/forest/releases/latest)
+[![dependency status](https://deps.rs/repo/github/ChainSafe/forest/status.svg?style=for-the-badge)](https://deps.rs/repo/github/ChainSafe/forest)
+[![forest book](https://img.shields.io/badge/doc-book-green?style=for-the-badge)](https://chainsafe.github.io/forest/)
```
Comment (hanabi1224) [inline README.md:9]: Add `[rustdoc@main](https://chainsafe.github.io/forest/rustdoc/)` later once this is merged

File: README.md [hunk 3]
```diff
@@ -205,8 +207,7 @@ You can detach Forest process via the `--detach` flag so that it runs in the bac
 The command will block until the detached Forest process has started its RPC server, allowing you to chain some RPC command immediately after.
 
 ### Documentation
-_Work in progress_.
-- https://chainsafe.github.io/forest/
+- [forest book (_Work in progress_)](https://chainsafe.github.io/forest/)
```
Comment (hanabi1224) [inline README.md:210]: Add `- [rust doc](https://chainsafe.github.io/forest/rustdoc/)` later once this is merged

**Review by elmattic — APPROVED** [review]

**Pull request merged**